pkg/exporter/sumologicexporter/exporter_test.go

```diff
@@ -740,25 +712,43 @@ func TestPushLogs_DontRemoveSourceAttributes(t *testing.T) {
 }
 
 func TestAllMetricsSuccess(t *testing.T) {
-	test := prepareExporterTest(t, createTestConfig(), []func(w http.ResponseWriter, req *http.Request){
-		func(w http.ResponseWriter, req *http.Request) {
-			body := extractBody(t, req)
-			expected := `test.metric.data{test="test_value",test2="second_value"} 14500 1605534165000
-gauge_metric_name{foo="bar",remote_name="156920",url="http://example_url"} 124 1608124661166
-gauge_metric_name{foo="bar",remote_name="156955",url="http://another_url"} 245 1608124662166`
-			assert.Equal(t, expected, body)
-			assert.Equal(t, "application/vnd.sumologic.prometheus", req.Header.Get("Content-Type"))
+	testcases := []struct {
+		name         string
+		expectedBody string
+		metricFunc   func() (pdata.Metric, pdata.Map)
+	}{
+		{
+			name:         "sum",
+			expectedBody: `test.metric.data{test="test_value",test2="second_value"} 14500 1605534165000`,
+			metricFunc:   exampleIntMetric,
 		},
-	})
-	test.exp.config.MetricFormat = PrometheusFormat
+		{
+			name: "gauge",
+			expectedBody: `` +
+				`gauge_metric_name{foo="bar",remote_name="156920",url="http://example_url"} 124 1608124661166` +
+				"\n" +
+				`gauge_metric_name{foo="bar",remote_name="156955",url="http://another_url"} 245 1608124662166`,
+			metricFunc: exampleIntGaugeMetric,
+		},
+	}
 
-	metrics := metricPairToMetrics([]metricPair{
-		exampleIntMetric(),
-		exampleIntGaugeMetric(),
-	})
```

Comment:
This test was design to verify sending multiple different metrics in one request AFAIR

Reply:
Could you add `sum and gauge` test as well?

Reply:
That test is kind of complex with the current setup and doesn't really make much practical sense imo.